File: tcpcom/Android/Examples/A_EchoClient/src/ch/aplu/echoclient/EchoClient.java
// EchoClient.java

package ch.aplu.echoclient;

import ch.aplu.android.*;
import ch.aplu.tcpcom.*;

public class EchoClient extends GameGrid implements TCPClientListener
{
  private GGConsole c;

  public void main()
  {
    String host = askIPAddress();
    int port = 5000;
    c = GGConsole.init();
    c.println("Hello. Trying to connect " + host);
    TCPClient client = new TCPClient(host, port);
    boolean success = client.connect(10);  // Timeout 10 s

    if (!success)
    {
      c.println("Connection failed");
      return;
    }
    client.addTCPClientListener(this);
    c.println("Connection established. Sending now...");
    for (int i = 0; i < 100; i++)
    {
      client.sendMessage("" + i);
      delay(100); // slow down
    }
    c.println("Program terminated");
    client.disconnect();
  }

  public void onStateChanged(String state, String msg)
  {
    if (state.equals(TCPClient.MESSAGE))
      c.println("Got: " + msg);
    else if (state.equals(TCPClient.DISCONNECTED))
      c.println("Disconnected");
  }

  private String askIPAddress()
  {
    GGPreferences prefs = new GGPreferences(this);
    String oldName = prefs.retrieveString("IPAddress");
    String newName = null;
    while (newName == null || newName.equals(""))
    {
      newName = GGInputDialog.show(this, "EV3", "Enter IP Address",
        oldName == null ? "10.0.1.1" : oldName);
    }
    prefs.storeString("IPAddress", newName);
    return newName;
  }
}

File: tcpcom/EV3/TigerJython/Examples/EchoClient.py
# EchoClient.py

from ev3robot import *
import time

def onStateChanged(state, msg):
    if state == TCPClient.MESSAGE:
        robot.drawString(msg, 0, 4)
    elif state == TCPClient.CONNECTED:
        robot.drawString("Connected", 0, 2)
    elif state == TCPClient.DISCONNECTED:
        robot.drawString("Disconnected", 0, 2)
 
robot = LegoRobot()
robot.drawString("Echo Client", 0, 1)
host = "192.168.0.17"
port = 5000
client = TCPClient(host, port, stateChanged = onStateChanged)
if client.connect():
    robot.drawString("Connected", 0, 2)
else:
    robot.drawString("Connection failed", 0, 2)
n = 0
while not robot.isEscapeHit():
    client.sendMessage(str(n))
    n += 1
    time.sleep(0.1)
client.disconnect()     
robot.exit()